tests/integration/test_ourlogs.py

```diff
@@ -87,37 +94,56 @@ def test_ourlog_extraction_with_otel_logs(
     envelope = envelope_with_otel_logs(start)
     relay.send_envelope(project_id, envelope)
 
-    ourlogs = ourlogs_consumer.get_ourlogs()
-    expected = {
-        "organization_id": 1,
-        "project_id": 42,
-        "retention_days": 90,
-        "timestamp_nanos": int(start.timestamp() * 1e9),
-        "observed_timestamp_nanos": time_within_delta(start, expect_resolution="ns"),
-        "trace_id": "5b8efff798038103d269b633813fc60c",
-        "body": "Example log record",
-        "trace_flags": 0,
-        "span_id": "eee19b7ec3c1b174",
-        "severity_text": "Information",
-        "severity_number": 10,
-        "received": time_within_delta(start, expect_resolution="s"),
-        "attributes": {
-            "string.attribute": {"string_value": "some string"},
-            "boolean.attribute": {"bool_value": True},
-            "int.attribute": {"int_value": 10},
-            "double.attribute": {"double_value": 637.704},
-            "sentry.severity_number": {"int_value": 10},
-            "sentry.severity_text": {"string_value": "Information"},
-            "sentry.timestamp_nanos": {
-                "string_value": str(int(start.timestamp() * 1e9))
-            },
-            "sentry.trace_flags": {"int_value": 0},
-        },
-    }
+    timestamp_nanos = round(start.timestamp() * 1e9)
+    timestamp_proto = Timestamp()
+    timestamp_proto.FromNanoseconds(timestamp_nanos)
+
+    expected_logs = [
+        MessageToDict(
+            TraceItem(
+                organization_id=1,
+                project_id=project_id,
+                timestamp=timestamp_proto,
+                trace_id="5b8efff798038103d269b633813fc60c",
+                item_id=timestamp_nanos.to_bytes(
+                    length=16,
+                    byteorder="little",
+                    signed=False,
+                ),
```

Comment:
Does this work? Should the uuidv7 have a random portion when `NoContext` is set? I think if it's set to 0 we may have collision problems

Reply:
This doesn't work, I couldn't find a proper `uuid7` library in Python to make the uuid based on a timestamp. That's why I replace it in the response lower.

Python 3.14 will have a function for this in the standard library.

Reply:
This is repeated because I'm using the `item_id` coming from Relay. We can't properly make a uuid7 from a custom timestamp in Python just yet so I am just ignoring this value.

The way the uuid7 is generated from the timestamp in Rust is correct though, and has this random portion.